lib/vm.js

```diff
@@ -39,6 +39,19 @@ const {
 // - isContext(sandbox)
 // From this we build the entire documented API.
 
+class Script extends ContextifyScript {
+  constructor(code, options) {
+    // Calling `ReThrow()` on a native TryCatch does not generate a new
+    // abort-on-uncaught-exception check. A dummy try/catch in JS land
+    // protects against that.
```

Comment:
@nodejs/v8 Is this a bug in V8?

Reply:
Yeah a ReThrow is not actually a new throw. It just "re-activates" the exception.

Reply:
@hashseed Does that mean one could get the right behavior by exiting the `TryCatch` scope, then using `isolate->ThrowException()`?

Reply:
Yeah. Though the message you get with this kind of "rethrow" will show the stack trace of the new throw location.